modules/trafficgateBidAdapter.js

```diff
@@ -1,95 +1,155 @@
+import {BANNER, VIDEO} from '../src/mediaTypes.js';
+import {ortbConverter} from '../libraries/ortbConverter/converter.js';
+import {deepAccess, mergeDeep, deepSetValue, convertTypes} from '../src/utils.js';
 
 const BIDDER_CODE = 'trafficgate';
-const URL = 'https://[HOST].bc-plugin.com/?c=o&m=multi'
+const URL = 'https://[HOST].bc-plugin.com/prebidjs'
 
 export const spec = {
   code: BIDDER_CODE,
-  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+  supportedMediaTypes: [BANNER, VIDEO],
+  isBidRequestValid,
+  buildRequests,
+  interpretResponse,
+  transformBidParams,
+  isBannerBid
+};
 
-  isBidRequestValid: function (bid) {
-    return !!(bid.bidId && bid.params && parseInt(bid.params.placementId) && bid.params.host)
-  },
-  /**
-   * Make a server request from the list of BidRequests.
-   *
-   * @param {BidRequest[]} validBidRequests A non-empty list of valid bid requests that should be sent to the Server.
-   * @return ServerRequest Info describing the request to the server.
-   */
-  buildRequests: function (validBidRequests) {
-    // convert Native ORTB definition to old-style prebid native definition
-    validBidRequests = convertOrtbRequestToProprietaryNative(validBidRequests);
-    if (validBidRequests && validBidRequests.length === 0) return [];
-
-    const location = getWindowLocation()
-    const placements = []
-
-    validBidRequests.forEach((bidReq) => {
-      placements.push({
-        placementId: bidReq.params.placementId,
-        bidId: bidReq.bidId,
-        traffic: getMediatype(bidReq)
-      })
-    })
+registerBidder(spec)
 
-    return {
-      method: 'POST',
-      url: URL.replace('[HOST]', validBidRequests[0].params.host),
-      data: {
-        language: (navigator && navigator.language) ? navigator.language : '',
-        secure: +(location.protocol === 'https:'),
-        host: location.hostname,
-        page: location.pathname,
-        placements: placements
+const converter = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 300
+  },
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+    mergeDeep(imp, {
+      ext: {
+        bidder: {
+          placementId: bidRequest.params.placementId,
+          host: bidRequest.params.host
+        }
       }
+    });
+    if (bidRequest.params.customFloor && !imp.bidfloor) {
+      imp.bidfloor = bidRequest.params.customFloor;
     }
+    return imp;
   },
-
-  interpretResponse: function (opts) {
-    const body = opts.body
-    const response = []
-
-    for (let i = 0; i < body.length; i++) {
-      const item = body[i]
-      if (isBidResponseValid(item)) {
-        response.push(item)
+  request(buildRequest, imps, bidderRequest, context) {
+    const req = buildRequest(imps, bidderRequest, context);
+    mergeDeep(req, {
+      at: 1,
+    })
+    const bid = context.bidRequests[0];
+    if (bid.params.coppa) {
+      deepSetValue(req, 'regs.coppa', 1);
+    }
+    if (bid.params.doNotTrack) {
```

Comment:
also this is something prebid willl set based on several checks here:
https://github.com/prebid/Prebid.js/blob/master/src/fpd/enrichment.js#L99
https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L938

So I think not overwriting is the right thing to do here

Reply:
I removed this too.